### ISSUE TITLE
unable to initialize statemachine with current state 

### ISSUE THREAD

**Issue by vidyli** [korken89/smlang-rs#19]
I am a littler curious why we  can't specify state when creating Statemachine, for example, we may load the state from a database ,  `Cart -> New -> Completed`.

```
 impl<T: StateMachineContext> StateMachine<T> {
            /// Creates a new state machine with the specified starting state.
            #[inline(always)]
            pub fn new(context: T) -> Self {
                StateMachine {
                    state: States::#starting_state,
                    context
                }
            }

          //...
}

```

**Comment by Yatekii** [comment]
I think this is just something that was not used yet :) I guess most state machines are run from the same initial state every time. You can easily have a "sytsem boot" state which then goes to the correct state immediately :)

I see the advantages of targeted state creation tho! So adding this would be welcome imo.

**Comment by korken89** [comment]
Fixed in #20
